lab_1/main.py

```diff
@@ -4,72 +4,79 @@
 """
 
 
-def tokenize(text: str) -> list:
-    """
-    Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
-    :param text: the initial text
-    :return: a list of lowercased tokens without punctuation
-    e.g. text = 'The weather is sunny, the man is happy.'
-    --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
-    """
-    pass
-
-
-def remove_stop_words(tokens: list, stop_words: list) -> list:
-    """
-    Removes stop words
-    :param tokens: a list of tokens
-    :param stop_words: a list of stop words
-    :return: a list of tokens without stop words
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
-    stop_words = ['the', 'is']
-    --> ['weather', 'sunny', 'man', 'happy']
-    """
-    pass
-
-
-def calculate_frequencies(tokens: list) -> dict:
-    """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens without stop words
-    :return: a dictionary with frequencies
-    e.g. tokens = ['weather', 'sunny', 'man', 'happy']
-    --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
-    """
-    pass
-
-
-def get_top_n_words(freq_dict: dict, top_n: int) -> list:
-    """
-    Returns the most common words
-    :param freq_dict: a dictionary with frequencies
-    :param top_n: a number of the most common words to return
-    :return: a list of the most common words
-    e.g. tokens = ['weather', 'sunny', 'man', 'happy', 'and', 'dog', 'happy']
-    top_n = 1
-    --> ['happy']
-    """
-    pass
-
-
-def get_concordance(tokens: list, word: str, left_context_size: int, right_context_size: int) -> list:
-    """
-    Gets a concordance of a word
-    A concordance is a listing of each occurrence of a word in a text,
-    presented with the words surrounding it
-    :param tokens: a list of tokens
-    :param word: a word-base for a concordance
-    :param left_context_size: the number of words in the left context
-    :param right_context_size: the number of words in the right context
-    :return: a concordance
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
-                    'the', 'dog', 'is', 'happy', 'but', 'the', 'cat', 'is', 'sad']
-    word = 'happy'
-    left_context_size = 2
-    right_context_size = 3
-    --> [['man', 'is', 'happy', 'the', 'dog', 'is'], ['dog', 'is', 'happy', 'but', 'the', 'cat']]
-    """
-    pass
+def tokenize(text):
+    if type(text) == str:
```

Comment:
*styling

better write `if not type(text) == str: return []` and implement main functionality after this check. Try to avoid nesting